src/chart/generateCategoricalChart.tsx

```diff
@@ -2170,5 +2176,14 @@ export const generateCategoricalChart = ({
         </ChartDataContextProvider>
       );
     }
+  }
+
+  return function CategoricalChart(props: CategoricalChartProps) {
+    const store = useRef(createStore(props.id ?? chartName));
+    return (
+      <Provider store={store.current}>
+        <CategoricalChartWrapper {...props} dispatch={store.current.dispatch} />
```

Comment:
This way the CategoricalChartWrapper has access to the store too, and can dispatch actions. This will allow us to slowly migrate things from its state to redux state.